src/factor_graph.cc

```diff
@@ -41,32 +43,85 @@ std::ostream &operator<<(std::ostream &stream,
   return stream;
 }
 
+template <class T>
+T *fast_alloc(size_t num) {
+  void *raw_memory = operator new[](num * sizeof(T));
+  return static_cast<T *>(raw_memory);
+}
+
 FactorGraph::FactorGraph(const FactorGraphDescriptor &capacity)
     : capacity(capacity),
       size(),
-      weights(new Weight[capacity.num_weights]),
-      factors(new Factor[capacity.num_factors]),
-      vifs(new FactorToVariable[capacity.num_edges]),
-      variables(new Variable[capacity.num_variables]),
-      factor_index(new size_t[capacity.num_edges]),
-      values(new VariableToFactor[capacity.num_values]) {}
+      // fast alloc: 0 sec for a 270M-factor graph
```

Comment:
AWESOME!  Seems like the magic is skipping initialization by constructors, and I'm trying to think if there's any place things might go wrong.  I guess this will be fine as long as we don't keep pointers in these data structs.  Not sure if there's an easy way to enforce that in C++ other than putting up some ALL-CAPS comments on those classes.  Maybe also rename it to `fast_alloc_no_init()`?